src/cargo/sources/registry/index.rs

```diff
@@ -592,17 +591,10 @@ impl<'cfg> RegistryIndex<'cfg> {
             let callback = &mut |s: IndexSummary| {
                 if !s.is_offline() {
                     called = true;
-                    f(s.into_summary());
+                    f(s);
                 }
             };
-            ready!(self.query_inner_with_online(
-                name,
-                req,
-                load,
-                yanked_whitelist,
-                callback,
-                false
-            )?);
+            ready!(self.query_inner_with_online(name, req, load, callback, false)?);
             if called {
```

Comment:
by moving filtering of yanked versions from beneath this code to above this code there may be situations in which `called` is no different. I don't think this will break anything, tests still pass, but it needs more thinking than I gave it.

Reply:
So we query offline and see if anything came up, if it did, we move on.  We then query online.

So I guess in theory if what the offline check found was yanked, then we might move on, filter it out, and then error instead of switching to on.  Whether that can actually happen and cause problems needs a wider view of this stack than I have at this time.  From how this is all behaving, I'm assuming it will be fine?

Reply:
That analysis sounds correct.

> (this should likely only happen with locked packages which will be allow-listed)

I think the problematic case is that there is a downloaded version that has been yanked and is not in the allowed list. The old behavior would filter out the version because it yanked  and not in the allow list, meaning that the first pass would see no candidates and so fallback to the second pass. Whereas, the new code will see the yanked version witch is available off-line so return it is the only candidate not falling back to the second pass.

Reply:
So it looks like we are allowing yanked in that first pass, which is fine because its a lockfile, bypassing this problem.  Do I have that right?